tests/unit/parserTest.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const tape = require('tape')
+const {
+  parseIncomingBody,
+  parseOutgoingBody
+} = require('../../src/middleware/parser')
+
+tape.test('Parser', t => {
+  t.test('parseIncomingBody', t => {
+    t.test(
+      'should throw error when transformation method is not supported',
+      async t => {
+        t.plan(1)
+        const ctx = {}
+        const inputFormat = 'unsupported'
+        const next = () => {}
+        try {
+          await parseIncomingBody(ctx, inputFormat, next)
+        } catch (error) {
+          t.equal(
+            error.message,
+            'transformation method "unsupported" not yet supported'
+          )
+        }
```

Comment:
I think instead of creating a `try/catch` block for checking the error is thrown, we could use the `.throws` method within tape to test this.
https://github.com/substack/tape#tthrowsfn-expected-msg

Reply:
t.throws only works for synchronous methods

Reply:
But with tap we can use t.rejects or t.throwsAsync